### PR TITLE
Runtime todo list/exploration

### PR DESCRIPTION
**Pull request by crufter**

### CODE REVIEW

File: ideas/runtime-todo.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- go.mod replaces to local packages work
+- access to private repo GitHub repos might also work because the `micro server` has access to the same envars as the user.
+
+A more useful reference point and though framework is the test suite: since they run on docker container there is a level of isolation.
+In fact the test suite is exactly what triggered the "uploading local code to `micro server`" functionality implementation as `micro run .` became utterly broken
+once the tests were ran in containers.
+
+We need to also think about both local codes (`micro run .`) and urls (`micro run github.com/micro/examples/helloworld`) when discussing features, as both should work.
+
+## Accessing private code
```
Comment (domwong) [inline ideas/runtime-todo.md:17]: For MVP do we care? Or is it enough to only support code that uses public repos for now?  @asim to make the call

File: ideas/runtime-todo.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- User acquires a [personal access token from GitHub](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) and does a `micro config set [key to be decided] [access token]`
+- The runtime can then use said token to access source code
+  - Default runtime might need to do specific setup steps because every machine is different [This article contains some ideas, although it is docker specific](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line)
+  - K8s runtime can have prebuilt images which handles all this given the token is passed down to it.*
+
+(*) This takes us to our next point
+
+## Running code in k8s without CI
+
+We have previously explored the concept of [dynamically downloading source code](https://github.com/micro/micro/tree/master/service/runtime/cells) and running it. The same idea could be done for v1 and it would closely match the spirit of the local workflow too: `micro update .` followed by `micro logs -f [name]` to see build output.
```
Comment (domwong) [inline ideas/runtime-todo.md:30]: What does this mean in practice? Where is code pulled from ? Does it come from my laptop? From Github? Somewhere else?

File: ideas/runtime-todo.md (hunk 1 above)
Comment (crufter) [reply ideas/runtime-todo.md:17]: I know I was not summoned but I would cast my vote for not caring or at least being lower prio than having the code actually running... If it's not running at all no point in accessing private repos.

File: ideas/runtime-todo.md (hunk 2 above)
Comment (crufter) [reply ideas/runtime-todo.md:30]: For `micro run [url]` it comes from GitHub (and other providers later).
For `micro run [local path]` it gets uploaded to store and somehow(tm) (to be defined by likely @jakexks) we should make it accessible to the container.

**Pull request merged**